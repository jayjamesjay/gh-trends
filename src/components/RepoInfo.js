/**
 * @module RepoInfo
 */

/**
 * Information about repository
 * @class
 */
class RepoInfo {
  /**
   * Creates a RepoInfo
   *
   * @param {string} nameWithOwner - name and author of repository
   * @param {string} url - url of the repository
   * @param {string} description - description
   * @param {number} stargazersCount - amount of stars
   * @param {string} language - programming language
   * @param {number} forks - amount of forks
   * @param {string} license - repository license
   */
  constructor(nameWithOwner, url, description, stargazersCount, language, forks, license) {
    this.nameWithOwner = nameWithOwner;
    this.url = url;
    this.description = description;
    this.stargazersCount = stargazersCount;
    this.language = language;
    this.forks = forks;
    this.license = license;
  }

  /**
   * Creates RepoInfo from Github API data
   *
   * @param {object} item - repository data from Github API
   * @returns {RepoInfo}
   */
  static fromGithubRes(item) {
    const { description, language, forks } = item;
    const license = item.license ? item.license.spdx_id : null;

    return new RepoInfo(
      item.full_name,
      item.html_url,
      description,
      item.stargazers_count,
      language,
      forks,
      license
    );
  }
}

export default RepoInfo;
