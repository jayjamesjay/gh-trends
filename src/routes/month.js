import React, { Component } from "react";
import StandarView from "../components/main";
import formatDate from "../components/date";

const monthAgo = new Date();
monthAgo.setMonth(monthAgo.getMonth() - 1);
const monthAgoStr = formatDate(monthAgo);

const query = `sort:stars-desc created:>2019-05-02`;
const title = "Trending repos this month";
const description = "Repositories created this month with biggest amount of stars."

export default class Month extends Component {
  render() {
    return <StandarView query={query} title={title} description={description}/>;
  }
}
