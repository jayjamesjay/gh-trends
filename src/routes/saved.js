import React, { Component } from "react";

export default class Month extends Component {
  render() {
    return <h1>Work in Progress...</h1>;
  }
}
