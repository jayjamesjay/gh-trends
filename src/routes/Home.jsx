import React, { useCallback, useEffect, useMemo } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import getAndSave, { Url, defApi, perPage, addLang, load } from '../components/Fetch';
import { ViewSingle } from '../components/View';
import { queries, languages } from '../components/Data';
import RepoInfoList from '../components/RepoInfoList';
import RepoInfo from '../components/RepoInfo';
import { H1 } from '../styles/Headers';
import { FormAlt } from '../styles/Form';
import Select from '../components/Select';
import { save } from '../actions';

const langs = ['All Languages', ...Object.keys(languages)];
const timeList = Object.keys(queries);

const mapStateToProps = state => {
  return {
    saved: state.saved
  };
};

const mapDispatchToProps = { save };

// eslint-disable-next-line no-shadow
export function Home({ saved, save }) {
  const [lang, setLang] = React.useState('All');
  const [time, setTime] = React.useState('This Week');
  const [repoInfo, setRepoInfo] = React.useState(new RepoInfoList([], 1));
  const [loading, setLoading] = React.useState(load.INPROGRESS);
  const abortController = new AbortController();
  const { signal } = abortController;
  const query = useMemo(() => addLang(queries[time], lang), [lang, time]);

  const onSelect = useCallback((event, callback) => {
    const val = event.target.value;
    callback(val);
  }, []);

  const selectLang = useCallback(
    event => {
      onSelect(event, setLang);
    },
    [onSelect, setLang]
  );

  const selectTime = useCallback(
    event => {
      onSelect(event, setTime);
    },
    [onSelect, setTime]
  );

  const loadData = useCallback(
    (currData = [], currPage = 1) => {
      setLoading(load.INPROGRESS);
      const preUrl = new Url(defApi).query(query).parts(perPage);
      const infoList = new RepoInfoList(currData, currPage);
      return getAndSave(infoList, preUrl, signal, setRepoInfo)
        .then(() => {
          setLoading(load.LOADED);
        })
        .catch(() => {
          setLoading(load.ERORR);
        });
    },
    [query, signal, setRepoInfo]
  );

  useEffect(() => {
    loadData();

    return () => {
      abortController.abort();
    };
  }, [lang, time]); // eslint-disable-line

  return (
    <>
      <header>
        <H1>Trending repositories</H1>
      </header>
      <FormAlt>
        <Select curr={lang} onSelect={selectLang} options={langs} label="Languages" />
        <Select curr={time} onSelect={selectTime} options={timeList} label="Periods of time" />
      </FormAlt>
      <ViewSingle
        data={repoInfo.data}
        loadData={() => loadData(repoInfo.data, repoInfo.page)}
        save={save}
        saved={saved}
        loading={loading}
      />
    </>
  );
}

Home.propTypes = {
  saved: PropTypes.arrayOf(PropTypes.instanceOf(RepoInfo)).isRequired,
  save: PropTypes.func.isRequired
};

const HomeContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);

export default HomeContainer;
