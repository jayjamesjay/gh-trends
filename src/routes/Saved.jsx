import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import View from '../components/View';
import Categories from '../components/Categories';
import { H1 } from '../styles/Headers';
import { ImgIcon } from '../styles/Img';
import { Bar } from '../styles/Main';
import { DownloadLink, jsonToMarkdown } from '../components/Data';
import RepoInfo from '../components/RepoInfo';
import { ButtonRemove } from '../styles/Button';
import { save, removeAllSaved } from '../actions';

const mapStateToProps = state => {
  return {
    saved: state.saved
  };
};

const mapDispatchToProps = { removeAllSaved, save };

// eslint-disable-next-line no-shadow
export function Saved({ saved, save, removeAllSaved }) {
  const [active, setActive] = React.useState('JSON');
  const imgDownload = <ImgIcon src="./assets/img/download.svg" alt="Download saved items" />;
  let content;

  if (saved.length > 0) {
    const link =
      active === 'JSON' ? (
        <DownloadLink
          filename="saved.json"
          content={JSON.stringify(saved)}
          dataType="json"
          display={imgDownload}
        />
      ) : (
        <DownloadLink filename="saved.md" content={jsonToMarkdown(saved)} display={imgDownload} />
      );

    content = (
      <>
        <header>
          <H1>Saved repositories</H1>
        </header>
        <section>
          <Bar>
            {link}
            <ButtonRemove onClick={removeAllSaved}>
              <ImgIcon src="./assets/img/remove-all.svg" alt="Remove all" />
            </ButtonRemove>
          </Bar>
          <Categories labels={['JSON', 'Markdown']} active={active} onClick={setActive} />
          <View data={saved} save={save} saved={saved} />
        </section>
      </>
    );
  } else {
    content = <H1>You haven&apos;t saved any repos...</H1>;
  }

  return content;
}

Saved.propTypes = {
  saved: PropTypes.arrayOf(PropTypes.instanceOf(RepoInfo)).isRequired,
  removeAllSaved: PropTypes.func.isRequired,
  save: PropTypes.func.isRequired
};

const SavedContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Saved);

export default SavedContainer;
