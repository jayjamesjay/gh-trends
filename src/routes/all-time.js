import React, { Component } from "react";
import { DataSquare } from "../components/main";

export default class AllTime extends Component {
  render() {
    return (
      <DataSquare
        title="All Time"
        content="Odio ut sem nulla pharetra diam sit. Ipsum dolor sit amet 
          consectetur adipiscing elit pellentesque habitant. Mattis aliquam 
          faucibus purus in massa tempor nec feugiat nisl. Hac habitasse platea 
          dictumst vestibulum rhoncus est. Adipiscing tristique risus nec feugiat. 
          Sed blandit libero volutpat sed cras. Interdum consectetur libero id 
          faucibus nisl tincidunt. Tristique senectus et netus et malesuada. 
          Vitae tempus quam pellentesque nec nam aliquam sem et. Semper feugiat 
          nibh sed pulvinar proin gravida hendrerit lectus a. Fusce ut placerat 
          orci nulla pellentesque. Eget magna fermentum iaculis eu non. Tellus 
          integer feugiat scelerisque varius morbi. Duis ut diam quam nulla. 
          In iaculis nunc sed augue."
      />
    );
  }
}
