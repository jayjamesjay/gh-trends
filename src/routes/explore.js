import React, { Component } from "react";
import { InfoBlock } from "../components/main";

//Temporarily added testing data
const dataExplore = {
  data: {
    search: {
      nodes: [
        {
          nameWithOwner: "imfunniee/gitfolio",
          url: "https://github.com/imfunniee/gitfolio",
          description:
            ":octocat: personal website + blog  for every github user",
          stargazers: {
            totalCount: 3124
          },
          repositoryTopics: {
            nodes: [
              {
                topic: {
                  name: "personal-website"
                }
              },
              {
                topic: {
                  name: "awesome"
                }
              },
              {
                topic: {
                  name: "node"
                }
              }
            ]
          },
          languages: {
            nodes: [
              {
                name: "HTML",
                color: "#e34c26"
              }
            ]
          }
        },
        {
          nameWithOwner: "nokia/memory-profiler",
          url: "https://github.com/nokia/memory-profiler",
          description: null,
          stargazers: {
            totalCount: 1624
          },
          repositoryTopics: {
            nodes: []
          },
          languages: {
            nodes: [
              {
                name: "Shell",
                color: "#89e051"
              }
            ]
          }
        }
      ]
    }
  }
};

const data = dataExplore.data.search.nodes;

export default class Explore extends Component {
  render() {
    return (
      <>
        {data.map((node, i) => (
          <InfoBlock key={node.nameWithOwner} info={node} />
        ))}
      </>
    );
  }
}
