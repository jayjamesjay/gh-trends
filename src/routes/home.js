import React, { Component } from 'react';
import getJSON, {
  requestUrl,
  api,
  order,
  sort,
  perPage,
  query,
} from '../components/fetch';
import { InfoBlock, Tabs } from '../components/main';
import { Content, ButtonMain } from '../components/styles';

import Data, { queryList, initData } from '../components/data';

export default class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [
        new Data('Week', initData, 1),
        new Data('Month', initData, 1),
        new Data('All Time', initData, 1),
      ],
    };
  }

  componentDidMount() {
    const newData = this.state.data.slice();
    const idList = ['Week', 'Month', 'All Time'];

    for (let i = 0; i < newData.length; i++) {
      const url = requestUrl(api, query(queryList[i]), sort, order, perPage);

      this._asyncRequest = getJSON(url).then(result => {
        this._asyncRequest = null;

        newData[i] = new Data(idList[i], Data.fromGithubRes(result.items), 1);

        this.setState({
          data: newData,
        });
      });
    }
  }

  componentWillUnmount() {
    if (this._asyncRequest) {
      this._asyncRequest.cancel();
    }
  }

  loadData = id => {
    const currData = this.state.data;
    const idx = currData.findIndex(elem => elem.id === id);
    const url = requestUrl(
      api,
      query(queryList[idx]),
      sort,
      order,
      perPage,
      'page=' + (currData[idx].page + 1)
    );

    getJSON(url).then(result => {
      currData[idx].data = currData[idx].data.concat(
        Data.fromGithubRes(result.items)
      );
      currData[idx].page += 1;

      this.setState({
        data: currData,
      });
    });
  };

  render() {
    return (
      <Tabs>
        {this.state.data.map(elem => {
          return (
            <div key={elem.id} label={elem.id}>
              <View id={elem.id} data={elem.data} loadData={this.loadData} />
            </div>
          );
        })}
      </Tabs>
    );
  }
}

export class View extends Component {
  onClick = () => {
    this.props.loadData(this.props.id);
  };

  render() {
    return (
      <>
        <Content>
          {this.props.data.map((node, id) => (
            <InfoBlock key={id} info={node} />
          ))}
        </Content>
        <footer>
          <ButtonMain onClick={this.onClick}>Show more</ButtonMain>
        </footer>
      </>
    );
  }
}
