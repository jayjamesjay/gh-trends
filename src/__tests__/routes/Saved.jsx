import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import { Saved } from '../../routes/Saved';
import RepoInfo from '../../utils/RepoInfo';

describe('<Saved />', () => {
  const func = () => {};
  const initData = new Array(6);
  const wrapper = () => render(<Saved save={func} removeAllSaved={func} saved={initData} />);

  for (let i = 0; i < initData.length; i += 1) {
    const curr = new RepoInfo(
      'jayjamesjay/gh-trends',
      '',
      'Loading content for this website...',
      123,
      'JavaScript',
      321,
      'MIT',
    );
    curr.nameWithOwner += i;
    initData[i] = curr;
  }

  it(`renders with default style`, () => {
    const component = renderer.create(<Saved save={func} removeAllSaved={func} saved={initData} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with title - no repos', () => {
    render(<Saved save={func} removeAllSaved={func} saved={[]} />);
    const title = `You haven't saved any repos...`;

    expect(screen.getByRole('heading').textContent).toContain(title);
  });

  it('renders with title - some repos', () => {
    wrapper();
    const title = `Saved repositories`;

    expect(screen.getByRole('heading', { level: 1 }).textContent).toContain(title);
  });

  it('creates default download link', () => {
    wrapper();
    const name = `saved.json`;

    expect(screen.getAllByRole('link')[0].getAttribute('download')).toEqual(name);
  });

  it('creates download link for Markdown', () => {
    const name = `saved.md`;
    render(<Saved save={func} removeAllSaved={func} saved={initData} />);
    fireEvent.click(screen.getByLabelText('Markdown'));

    expect(screen.getAllByRole('link')[0].getAttribute('download')).toEqual(name);
  });

  it('onClick', () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((init) => [init, setState]);

    render(<Saved save={func} removeAllSaved={func} saved={initData} />);
    fireEvent.click(screen.getByLabelText('Markdown'));

    expect(setState).toHaveBeenCalledWith('Markdown');
    jest.clearAllMocks();
  });
});
