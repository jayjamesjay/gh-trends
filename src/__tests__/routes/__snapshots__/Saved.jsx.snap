// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Saved /> renders with default style 1`] = `
<DocumentFragment>
  .c0 {
  margin: 3rem auto 1rem auto;
  font-size: 2rem;
  text-transform: uppercase;
}

@media screen and (min-width:40rem) {

}

@media screen and (min-width:40rem) {

}

<header>
    <h1
      class="c0"
    >
      Saved repositories
    </h1>
  </header>
  .c0 {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-content: center;
  align-items: center;
}

.c15 {
  margin: auto 0 0 0;
}

.c4 {
  cursor: pointer;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: 0;
}

.c5 {
  display: flex;
  width: 3rem;
  height: 3rem;
  margin: -1.2rem 0 0 0;
  padding: 0;
}

.c16 {
  margin: 0.5rem 0 0 0;
}

.c1 {
  display: block;
  font-size: 1.2rem;
  font-family: 'Noto Sans HK',sans-serif;
  color: #000;
  text-decoration: none;
  cursor: pointer;
}

.c12 {
  display: flex;
  margin: 2rem 1rem;
  min-height: 13rem;
  padding: 1.2rem 1.2rem 0.8rem 1.2rem;
  flex: 1 1 100%;
  flex-flow: column nowrap;
  background: linear-gradient( #fff 0%,#fff 0.3rem,#f1e05a 0.3rem,#f1e05a 0.6rem,#fff 0.6rem );
  box-shadow: 0px 3px 5px 1px rgba(217,217,217,0.82);
  border-radius: 0.5rem;
  text-align: left;
  word-break: break-word;
}

.c14 {
  font-weight: 400;
  font-size: 1.3rem;
  color: #767676;
}

.c8 {
  position: relative;
  z-index: 100;
}

.c13 {
  margin: 0.2rem 0;
  font-size: 1.7rem;
}

.c2 {
  width: 100%;
  height: auto;
  filter: invert(0%);
}

.c3 {
  margin-top: 0.5rem;
  width: auto;
  height: 3rem;
}

.c19 {
  display: inline-block;
  width: 1.5rem;
  height: 1.5rem;
  filter: invert(0%);
}

.c17 {
  margin: 0 0.5rem;
  padding: 0;
}

.c18 {
  display: flex;
  align-content: center;
  align-items: center;
  margin: 0 0.5rem;
  padding: 0;
}

.c6 {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  width: 100%;
  text-transform: uppercase;
}

.c9 {
  opacity: 0;
}

.c7 {
  display: inherit;
  padding: 1rem 1.5rem;
  position: relative;
  background: transparent;
  overflow: hidden;
  font-size: 1.3rem;
  cursor: pointer;
}

.c7::before {
  content: '';
  width: 90%;
  height: 90%;
  position: absolute;
  top: 5%;
  left: 5%;
  z-index: 20;
  background: #fff;
  border-radius: 0.5rem;
  box-shadow: 0px 3px 5px 1px rgba(217,217,217,0.82);
  transition: transform 0.15s ease-in-out;
  transform: translateX(0%);
}

.c10 {
  display: inherit;
  padding: 1rem 1.5rem;
  position: relative;
  background: transparent;
  overflow: hidden;
  font-size: 1.3rem;
  cursor: pointer;
}

.c10::before {
  content: '';
  width: 90%;
  height: 90%;
  position: absolute;
  top: 5%;
  left: 5%;
  z-index: 20;
  background: #fff;
  border-radius: 0.5rem;
  box-shadow: 0px 3px 5px 1px rgba(217,217,217,0.82);
  transition: transform 0.15s ease-in-out;
  transform: translateX(-200%);
}

@media screen and (min-width:40rem) {
  .c11 {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-around;
    margin: 0 1.5rem;
  }
}

@media screen and (min-width:40rem) {
  .c12 {
    min-width: 16.5rem;
    margin: 1.5rem 1rem;
    flex: 1 1 27%;
  }
}

<section>
    <div
      class="c0"
    >
      <a
        class="c1"
        download="saved.json"
        href="data:json/plain;charset=utf-8,%5B%7B%22nameWithOwner%22%3A%22jayjamesjay%2Fgh-trends0%22%2C%22url%22%3A%22%22%2C%22description%22%3A%22Loading%20content%20for%20this%20website...%22%2C%22stargazersCount%22%3A123%2C%22language%22%3A%22JavaScript%22%2C%22forks%22%3A321%2C%22license%22%3A%22MIT%22%7D%2C%7B%22nameWithOwner%22%3A%22jayjamesjay%2Fgh-trends1%22%2C%22url%22%3A%22%22%2C%22description%22%3A%22Loading%20content%20for%20this%20website...%22%2C%22stargazersCount%22%3A123%2C%22language%22%3A%22JavaScript%22%2C%22forks%22%3A321%2C%22license%22%3A%22MIT%22%7D%2C%7B%22nameWithOwner%22%3A%22jayjamesjay%2Fgh-trends2%22%2C%22url%22%3A%22%22%2C%22description%22%3A%22Loading%20content%20for%20this%20website...%22%2C%22stargazersCount%22%3A123%2C%22language%22%3A%22JavaScript%22%2C%22forks%22%3A321%2C%22license%22%3A%22MIT%22%7D%2C%7B%22nameWithOwner%22%3A%22jayjamesjay%2Fgh-trends3%22%2C%22url%22%3A%22%22%2C%22description%22%3A%22Loading%20content%20for%20this%20website...%22%2C%22stargazersCount%22%3A123%2C%22language%22%3A%22JavaScript%22%2C%22forks%22%3A321%2C%22license%22%3A%22MIT%22%7D%2C%7B%22nameWithOwner%22%3A%22jayjamesjay%2Fgh-trends4%22%2C%22url%22%3A%22%22%2C%22description%22%3A%22Loading%20content%20for%20this%20website...%22%2C%22stargazersCount%22%3A123%2C%22language%22%3A%22JavaScript%22%2C%22forks%22%3A321%2C%22license%22%3A%22MIT%22%7D%2C%7B%22nameWithOwner%22%3A%22jayjamesjay%2Fgh-trends5%22%2C%22url%22%3A%22%22%2C%22description%22%3A%22Loading%20content%20for%20this%20website...%22%2C%22stargazersCount%22%3A123%2C%22language%22%3A%22JavaScript%22%2C%22forks%22%3A321%2C%22license%22%3A%22MIT%22%7D%5D"
      >
        <img
          alt="Download saved items"
          class="c2 c3"
          src="test-file-stub"
        />
      </a>
      <button
        class="c4 c5"
      >
        <img
          alt="Remove all"
          class="c2 c3"
          src="test-file-stub"
        />
      </button>
    </div>
    <form
      class="c6"
    >
      <div
        aria-checked="true"
        role="radio"
        tabindex="0"
      >
        <label
          class="c7"
          for="JSON"
        >
          <span
            class="c8"
          >
            JSON
          </span>
        </label>
        <input
          checked=""
          class="c9"
          id="JSON"
          tabindex="-1"
          type="radio"
          value="JSON"
        />
      </div>
      <div
        aria-checked="false"
        role="radio"
        tabindex="0"
      >
        <label
          class="c10"
          for="Markdown"
        >
          <span
            class="c8"
          >
            Markdown
          </span>
        </label>
        <input
          class="c9"
          id="Markdown"
          tabindex="-1"
          type="radio"
          value="Markdown"
        />
      </div>
    </form>
    <div
      class="c11"
    >
      <article
        class="c12"
      >
        <h2
          class="c13"
        >
          <a
            class="c1"
            href=""
          >
            <span
              class="c14"
            >
              jayjamesjay
            </span>
             /gh-trends0
          </a>
        </h2>
        <p>
          Loading content for this website...
        </p>
        <div
          class="c0 c15"
        >
          <button
            class="c4 c16"
          >
            <img
              alt="Remove from saved"
              class="c2"
              src="test-file-stub"
              title="Remove from saved"
            />
          </button>
          <p
            class="c17"
          >
            Forks: 321
          </p>
          <p
            class="c17"
          >
            License: MIT
          </p>
          <p
            class="c18"
          >
            123
            <img
              alt="Stars"
              class="c19"
              src="test-file-stub"
            />
          </p>
          <p
            class="c17"
          >
            Language: JavaScript
          </p>
        </div>
      </article>
      <article
        class="c12"
      >
        <h2
          class="c13"
        >
          <a
            class="c1"
            href=""
          >
            <span
              class="c14"
            >
              jayjamesjay
            </span>
             /gh-trends1
          </a>
        </h2>
        <p>
          Loading content for this website...
        </p>
        <div
          class="c0 c15"
        >
          <button
            class="c4 c16"
          >
            <img
              alt="Remove from saved"
              class="c2"
              src="test-file-stub"
              title="Remove from saved"
            />
          </button>
          <p
            class="c17"
          >
            Forks: 321
          </p>
          <p
            class="c17"
          >
            License: MIT
          </p>
          <p
            class="c18"
          >
            123
            <img
              alt="Stars"
              class="c19"
              src="test-file-stub"
            />
          </p>
          <p
            class="c17"
          >
            Language: JavaScript
          </p>
        </div>
      </article>
      <article
        class="c12"
      >
        <h2
          class="c13"
        >
          <a
            class="c1"
            href=""
          >
            <span
              class="c14"
            >
              jayjamesjay
            </span>
             /gh-trends2
          </a>
        </h2>
        <p>
          Loading content for this website...
        </p>
        <div
          class="c0 c15"
        >
          <button
            class="c4 c16"
          >
            <img
              alt="Remove from saved"
              class="c2"
              src="test-file-stub"
              title="Remove from saved"
            />
          </button>
          <p
            class="c17"
          >
            Forks: 321
          </p>
          <p
            class="c17"
          >
            License: MIT
          </p>
          <p
            class="c18"
          >
            123
            <img
              alt="Stars"
              class="c19"
              src="test-file-stub"
            />
          </p>
          <p
            class="c17"
          >
            Language: JavaScript
          </p>
        </div>
      </article>
      <article
        class="c12"
      >
        <h2
          class="c13"
        >
          <a
            class="c1"
            href=""
          >
            <span
              class="c14"
            >
              jayjamesjay
            </span>
             /gh-trends3
          </a>
        </h2>
        <p>
          Loading content for this website...
        </p>
        <div
          class="c0 c15"
        >
          <button
            class="c4 c16"
          >
            <img
              alt="Remove from saved"
              class="c2"
              src="test-file-stub"
              title="Remove from saved"
            />
          </button>
          <p
            class="c17"
          >
            Forks: 321
          </p>
          <p
            class="c17"
          >
            License: MIT
          </p>
          <p
            class="c18"
          >
            123
            <img
              alt="Stars"
              class="c19"
              src="test-file-stub"
            />
          </p>
          <p
            class="c17"
          >
            Language: JavaScript
          </p>
        </div>
      </article>
      <article
        class="c12"
      >
        <h2
          class="c13"
        >
          <a
            class="c1"
            href=""
          >
            <span
              class="c14"
            >
              jayjamesjay
            </span>
             /gh-trends4
          </a>
        </h2>
        <p>
          Loading content for this website...
        </p>
        <div
          class="c0 c15"
        >
          <button
            class="c4 c16"
          >
            <img
              alt="Remove from saved"
              class="c2"
              src="test-file-stub"
              title="Remove from saved"
            />
          </button>
          <p
            class="c17"
          >
            Forks: 321
          </p>
          <p
            class="c17"
          >
            License: MIT
          </p>
          <p
            class="c18"
          >
            123
            <img
              alt="Stars"
              class="c19"
              src="test-file-stub"
            />
          </p>
          <p
            class="c17"
          >
            Language: JavaScript
          </p>
        </div>
      </article>
      <article
        class="c12"
      >
        <h2
          class="c13"
        >
          <a
            class="c1"
            href=""
          >
            <span
              class="c14"
            >
              jayjamesjay
            </span>
             /gh-trends5
          </a>
        </h2>
        <p>
          Loading content for this website...
        </p>
        <div
          class="c0 c15"
        >
          <button
            class="c4 c16"
          >
            <img
              alt="Remove from saved"
              class="c2"
              src="test-file-stub"
              title="Remove from saved"
            />
          </button>
          <p
            class="c17"
          >
            Forks: 321
          </p>
          <p
            class="c17"
          >
            License: MIT
          </p>
          <p
            class="c18"
          >
            123
            <img
              alt="Stars"
              class="c19"
              src="test-file-stub"
            />
          </p>
          <p
            class="c17"
          >
            Language: JavaScript
          </p>
        </div>
      </article>
    </div>
  </section>
</DocumentFragment>
`;
